fabfile/add.py

```diff
@@ -2,119 +2,230 @@
+        test_name = self.test_name
+
+        dep_order = config_parser.get(test_name, 'dep_order').split(',')
+        for dep in dep_order:
+            dep_ext = Extension(dep)
+            dep_ext.parse_from_config(config_parser)
+            self.dep_exts.append(dep_ext)
+
+        self.test_command = config_parser.get(test_name, 'test_command')
+
+        if config_parser.has_option(test_name, 'conf_string'):
+            self.conf_lines = eval(config_parser.get(test_name, 'conf_string')).split()
+
+        extension_path = ''
+        if self.extension.contrib:
+            extension_path = contrib_path_for_extension(self.extension.name)
+        else:
+            extension_path = extension_path_for_url(self.extension.repo_url)
+        self.test_path = os.path.join(extension_path, self.extension.relative_test_path)
+
+    def get_install_tasks(self):
+        extension_install_tasks = []
+        for dept_ext in self.dep_exts:
+            extension_install_task = InstallExtensionTask(dept_ext)
+            extension_install_tasks.append(extension_install_task)
+        return extension_install_tasks
+
+    def get_configure_task(self):
+        return ConfigureExtensionTest(self.dep_exts, self.conf_lines)
+
+    def create_depended_extensions(self):
+        # add --load-extension=deps to REGRESS_OPTS in Makefile
+        # that will cause pg_regress to create those extensions which should be already installed (and preloaded if necessary)
+        with cd(self.test_path):
+            for dep_ext in self.dep_exts:
+                if dep_ext.create:
+                    run("echo 'REGRESS_OPTS := $(REGRESS_OPTS) --load-extension={}' >> Makefile".format(dep_ext.name))
+
+    def regress(self):
+        # force drop contrib_regression or regression dbs if exists, some backend still use db, so drop db not works
+        utils.psql("DROP DATABASE IF EXISTS contrib_regression WITH (FORCE);")
+        utils.psql("DROP DATABASE IF EXISTS regression WITH (FORCE);")
+
+        # create result folder if not exists
+        utils.mkdir_if_not_exists(config.RESULTS_DIRECTORY)
+
+        # load depended extensions
+        self.create_depended_extensions()
+
+        # run tests with warn_only option to not exit in case of a test failure in the extension,
+        # because we want to run regression tests for other extensions too.
+        with settings(warn_only=True):
+            with cd(self.test_path):
+                run(self.test_command)
+
+        # rename regression.diffs, if exists, so that extension's diff file does not conflict with others'        
+        self.rename_regression_diff()
+
+    def rename_regression_diff(self):
+        regression_diffs_path = os.path.join(self.test_path, config.REGRESSION_DIFFS_FILE)
+        if os.path.isfile(regression_diffs_path):
+            extension_regression_diff = config.REGRESSION_DIFFS_FILE + '_{}_'.format(self.test_name) + self.extension.name
```

Comment:
Would this name be unique for running with multiple postgres versions as well?
If not, maybe add the pg version to the name.